Components/grade-table.js

```diff
@@ -1,24 +1,49 @@
 class GradeTable {
-  constructor(tableElement) {
+  constructor(tableElement, noGradesElement) {
     this.tableElement = tableElement;
+    this.noGradesElement = noGradesElement;
   }
 
   updateGrades(grades) {
     console.log(grades);
     var tBody = $("#table");
     tBody.empty();
     for (var i = 0; i < grades.length; i++) {
-      var row = document.createElement("tr");
-      var studentName = document.createElement("td");
-      var course = document.createElement("td");
-      var grade = document.createElement("td");
-      tBody.append(row);
-      row.appendChild(studentName);
-      studentName.textContent = grades[i].name;
-      row.appendChild(course);
-      course.textContent = grades[i].course;
-      row.appendChild(grade);
-      grade.textContent = grades[i].grade;
+      this.renderGradeRow(grades[i], this.deleteGrade);
```

Comment:
Take the `<tr>` returned by `this.renderGradeRow()` and append it to `this.tableElement`.